src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingScopeHttpMessageHandler.cs

```diff
@@ -106,12 +109,16 @@ public static class EventIds
 
             public static IDisposable BeginRequestPipelineScope(ILogger logger, HttpRequestMessage request)
             {
-                return _beginRequestPipelineScope(logger, request.Method, request.RequestUri);
+                return _beginRequestPipelineScope(logger, request.Method, request.RequestUri.AbsoluteUri);
             }
 
             public static void RequestPipelineStart(ILogger logger, HttpRequestMessage request, Func<string, bool> shouldRedactHeaderValue)
             {
-                _requestPipelineStart(logger, request.Method, request.RequestUri, null);
+                // We check here to avoid allocating on the request.RequestUri.AbsoluteUri call unnecessarily
+                if (logger.IsEnabled(LogLevel.Information))
+                {
+                    _requestPipelineStart(logger, request.Method, request.RequestUri.AbsoluteUri, null);
+                }
```

Comment:
For the scope handler, doing the check just in `RequestPipelineStart` isn't sufficient since `BeginRequestPipelineScope` will allocate the `AbsoluteUri` anyway.

I think it's fine to drop the `IsEnabled` check here.

Reply:
Updated 👍